applications/luci-app-sshtunnel/htdocs/luci-static/resources/view/sshtunnel.js

```diff
@@ -0,0 +1,403 @@
+		o.rmempty = false;
+
+
+		s = m.section(form.GridSection, "_keys");
+		s.render = L.bind(_renderSshKeys, this, sshKeys);
+
+		s = m.section(form.GridSection, "_known_hosts");
+		s.render = L.bind(_renderKnownHosts, this, knownHosts);
+
+		return m.render();
+	},
+});
+
+function _addServerOption(s) {
+	var o = s.option(form.ListValue, "server", _("Server"), _("Type a server name from servers list above"));
+	o.datatype = "uciname";
+	o.rmempty = false;
+	uci.sections('sshtunnel', 'server', function (s, sectionName) {
+		o.value(sectionName, s.hostname ? '%s (%s)'.format(sectionName, s.hostname) : sectionName);
+	});
+	return o;
+}
+
+function _splitSshKeys(sshFiles) {
+	var sshKeys = {};
+	for (var i = 0; i < sshFiles.length; i++) {
+		var sshPubKeyName = sshFiles[i];
+		var sshKeyName = sshPubKeyName.substring(0, sshPubKeyName.length - 4);
+		i++;
+		var sshPub = sshFiles[i];
+		sshKeys[sshKeyName] = "<small><code>" + sshPub + "</code></small>";
+	}
+	allSshKeys = sshKeys;
+	return sshKeys;
+}
+
+function _renderSshKeys(sshKeys) {
+	var table = E('table', {'class': 'table cbi-section-table', 'id': 'keys_table'}, [
+		E('tr', {'class': 'tr table-titles'}, [
+			E('th', {'class': 'th'}, _('Name')),
+			E('th', {'class': 'th'}, _('Public Key')),
+		])
+	]);
+
+	var rows = Object.entries(sshKeys);
+	cbi_update_table(table, rows, E('em', _("No any SSH key, please generate a new one")));
+
+	var keyGenBtn = E('div', {}, [
+		E('label', {}, _('Generate a new key') + ': '),
+		E('span', {'class': 'control-group'}, [
+			E('input', {
+				'type': 'text',
+				'name': 'keyName',
+				'value': 'id_ed25519',
+				'keydown': function (ev) {
+					if (ev.keyCode === 13) _handleKeyGen(ev);
+				}
+			}),
+			E('button', {
+				'class': 'btn cbi-button cbi-button-action',
+				'click': _handleKeyGen
+			}, [_('Generate')])
+		])
+	]);
+	return E('div', {'class': 'cbi-section cbi-tblsection'}, [
+		E('h3', _('Keys')),
+		E('div', {'class': 'cbi-section-descr'},
+			_("You need to add the pub key to /root/.ssh/authorized_keys or to /etc/dropbear/authorized_keys.") + " " +
+			_("In LuCI you can do that with System / Administration / SSH-Keys")
+		),
+		keyGenBtn, table
+	]);
+}
+
+function _handleKeyGen(event) {
+	var keyName = document.querySelector('input[name="keyName"]').value;
+	if (!keyName) {
+		alert(_("Please specify a key name"));
+		return;
+	}
+	if (allSshKeys[keyName]) {
+		alert(_("A key with the name already exists"));
+		return;
+	}
+	if (/[^a-zA-Z0-9_.]/.test(keyName)) {
+		alert(_("Bad name"));
+		return;
+	}
+
+	let command = "/usr/bin/ssh-keygen";
+	let commandArgs = ["-t", "ed25519", "-q", "-N", "", "-f", "/root/.ssh/" + keyName];
+	if (!hasSshKeygen) {
+		command = "/usr/bin/dropbearkey";
+		commandArgs = ["-t", "ed25519", "-f", "/root/.ssh/" + keyName];
+	}
+	return fs.exec(command, commandArgs).then(function (res) {
+		if (res.code === 0) {
+			alert("A new key was generated. Please refresh the page in a browser to see it.");
```

Comment:
Just `return;` - handle failures, not success.

Reply:
Here I need to say "Please refresh the page in a browser to see it" otherwise a user won't see it

Reply:
I decided just reload the page instead.